Geometry isGeosValid to validateGeometry

matakuka opened the pull request:
On complicated polygons there can be a meeting of "inner ring" and "outer ring". IsGeosValid() returns False in these cases, causing issues like https://hub.qgis.org/issues/14532 . validateGeometry accepts the meeting of inner and outer rings.  I also inserted a print  to console to check what exactly was wrong.

jef-n committed:
2to3:
* add fixes
* pyqt fixer: include some QtSql symbols and remove imports
  which only symbol was removed (eg. SIGNAL)

Commit:
Geometry isGeosValid to validateGeometry

On complicated polygons there can be a meeting of "inner ring" and "outer ring". IsGeosValid() returns False in these cases, causing issues like https://hub.qgis.org/issues/14532 . validateGeometry accepts the meeting of inner and outer rings.  I also inserted a print  to console to check what exactly was wrong.

m-kuhn commented on python/plugins/processing/algs/qgis/Dissolve.py:
```diff
@@ -86,13 +86,18 @@ def processAlgorithm(self, progress):
+                    errors = tmpInGeom.validateGeometry()
+                    if len(errors) != 0:
+                        for error in errors:
+                            print error.what()
```
The output should be sent to the log instead of printing it. E.g. https://github.com/matakuka/QGIS/blob/patch-1/python/plugins/processing/algs/qgis/Intersection.py#L111

m-kuhn commented on python/plugins/processing/algs/qgis/Dissolve.py:
```diff
@@ -86,13 +86,18 @@ def processAlgorithm(self, progress):
                 else:
                     tmpInGeom = QgsGeometry(inFeat.geometry())
-                    if tmpInGeom.isGeosEmpty() or not tmpInGeom.isGeosValid():
+                    if tmpInGeom.isGeosEmpty():
```
Does the geometry not need to be validated here?

matakuka replied:
Sorry, newbie error. Fixed.

m-kuhn commented:
Thanks a lot for providing the patch. Making the algorithms more verbose in case of trouble and accept more input is always good.

It would be great if you could also add one or two test cases for the algorithm to make sure it acts nicely on NULL geometries and on the case which is fixed here so we don't run into any regressions down the road.

Commit:
Update Dissolve.py

second validation (was missing); error logging to ProcessingLog instead of stdout.

m-kuhn commented on python/plugins/processing/algs/qgis/Dissolve.py:
```diff
@@ -119,7 +136,7 @@ def processAlgorithm(self, progress):
                 tempItem = attrs[fieldIdx]
                 tmpInGeom = QgsGeometry(inFeat.geometry())
-                if tmpInGeom.isGeosEmpty() or not tmpInGeom.isGeosValid():
+                if tmpInGeom.isGeosEmpty():
```
It probably also applies to this one here? ;)

Commit:
Update Dissolve.py

Another missing geometry check reinserted.

matakuka commented:
I am not sure how to add "test cases" (still very new here). I checked it with the bug's valid polygons (both with Dissolve All and Dissolve by Field); and several line and polygon layers, with (a) valid geometries (b) valid and invalid geometries mixed and (c) valid and NULL geometries mixed. The output dissolves all features with a valid geometry and dumps  the invalid ones, generating a detailed log (what the error is and where, if that applies) to processingLog, as requested.

jef-n committed:
db_manager: cleanups

jef-n committed:
GdalTools: new style signals

jef-n committed:
processing: switch to pyqt wrappers, flake8 fixes

jef-n committed:
python console: switch to pyqt wrappers and new style signals, flake8 fixes, show exceptions in console also with pyqt5

jef-n committed:
metasearch: switch to pyqt wrappers

jef-n committed:
plugin installer: switch to pyqt wrappers

jef-n committed:
scripts/mkuidefaults.py: switch to PyQt wrappers and reformat

jef-n committed:
scripts/widgets_tree.py: switch to PyQt wrappers and reformat

jef-n committed:
switch tests to pyqt wrappers

jef-n committed:
switch bindings to pyqt wrappers

jef-n committed:
[FEATURE] oracle provider: basic support for Oracle Workspace Manager
(switches to a given workspace on connect)

jef-n committed:
travis: disable fcgi and server on osx (fastcgi.com apparently gone)

m-kuhn commented:
The instructions for processing tests can be found here. A lot of it is just some clicks in the user interface. You will probably want to add the data where the inner ring touches the outer one in python/plugins/processing/tests/testdata/custom.

https://github.com/qgis/QGIS/blob/master/python/plugins/processing/tests/README.md

Commit:
Tests for changes in dissolve.py

Commit:
Merge branch 'patch-1' of https://github.com/matakuka/QGIS

matakuka commented:
Sorry, I'll try again with a clean pull request. I did mess up somewhere.
